### ISSUE TITLE
Shape fill with custom pattern

### ISSUE THREAD

**Issue by mariusa** [GraphiteEditor/Graphite#950]
Hi,

I saw it's possible to fill a shape with a solid or gradient color.

Could custom patterns be supported too?

Here's a screenshot from Excalidraw:
![Screenshot from 2023-01-08 15-21-18](https://user-images.githubusercontent.com/1283842/211198607-d5303303-b370-449e-b064-b1fad54b9788.png)

Thanks!

**Comment by Keavon** [comment]
This will be possible with the node system in the future by generating any kind of pattern you'd like.

**Comment by 0HyperCube** [comment]
> This will be possible with the node system in the future by generating any kind of pattern you'd like.

Now we have the node system we probably need to think about how we add this to the vector data format.

**Comment by Keavon** [comment]
Yup! One consideration is just the rendering aspect (Vello and SVG, the latter of which will require SVG's masking feature). The other consideration is perhaps how to encode the repeating pattern starting point. Also, merging this conceptually with gradients would be useful too.

**Comment by 0HyperCube** [comment]
Note that in svg there is the pattern element https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Patterns

**Comment by Keavon** [comment]
Oh yes, good point.

**Comment by Keavon** [comment]
Closing in favor of #2779.
